azurerm/resource_arm_mariadb_virtual_network_rule.go

```diff
@@ -0,0 +1,242 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/mariadb/mgmt/2018-06-01/mariadb"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMariaDbVirtualNetworkRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMariaDbVirtualNetworkRuleCreateUpdate,
+		Read:   resourceArmMariaDbVirtualNetworkRuleRead,
+		Update: resourceArmMariaDbVirtualNetworkRuleCreateUpdate,
+		Delete: resourceArmMariaDbVirtualNetworkRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.VirtualNetworkRuleName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"server_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"subnet_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+		},
+	}
+}
+
+func resourceArmMariaDbVirtualNetworkRuleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadb.VirtualNetworkRulesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	serverName := d.Get("server_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	subnetId := d.Get("subnet_id").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, serverName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing MariaDb Virtual Network Rule %q (MariaDb Server: %q, Resource Group: %q): %+v", name, serverName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_mariadb_virtual_network_rule", *existing.ID)
+		}
+	}
+
+	// due to a bug in the API we have to ensure the Subnet's configured correctly or the API call will timeout
+	// BUG: https://github.com/Azure/azure-rest-api-specs/issues/3719
```

Comment:
according to the link this was closed in nov 2018?

Reply:
This is a direct copy of the MySQL version of this code, with relevant API calls replaced. Because that comment references PostgresSQL, I imagine the MySQL version was implemented the same way -- a copy. I imagine the tests will require the same (minimal) effort, as will the documentation. We are already using this to provision MariaDB servers

Reply:
Most likey, however it does indicate this is no longer needed. So could we remove the workaround in this PR as it seems like its no longer required?

Reply:
@katbyte, I thought about this some more, and I'm not comfortable taking this out for a couple of reasons:

- If the code is ever refactored to avoid duplication (and it should be), it will make things more difficult for the person making the change because it introduces an additional difference that needs to be inspected.
- The behavior here is reasonable. It's a sanity check for making sure the subnet exists, and it'll 404 if it doesn't with a clear and easy to understand error.

Reply:
That's a fair response, we'll investigate this & the other resources separately from this PR.